test/scenarios.py

```diff
@@ -2664,6 +2664,204 @@
+                            norddrop.QueuedFile(FILES["testfile-bulk-03"].id, "testfile-bulk-03", 10485760, "/tmp"),
+                            norddrop.QueuedFile(FILES["testfile-bulk-04"].id, "testfile-bulk-04", 10485760, "/tmp"),
+                            norddrop.QueuedFile(FILES["testfile-bulk-05"].id, "testfile-bulk-05", 10485760, "/tmp"),
+                            norddrop.QueuedFile(FILES["testfile-bulk-06"].id, "testfile-bulk-06", 10485760, "/tmp"),
+                            norddrop.QueuedFile(FILES["testfile-bulk-07"].id, "testfile-bulk-07", 10485760, "/tmp"),
+                            norddrop.QueuedFile(FILES["testfile-bulk-08"].id, "testfile-bulk-08", 10485760, "/tmp"),
+                            norddrop.QueuedFile(FILES["testfile-bulk-09"].id, "testfile-bulk-09", 10485760, "/tmp"),
+                            norddrop.QueuedFile(FILES["testfile-bulk-10"].id, "testfile-bulk-10", 10485760, "/tmp"),
+                        ]),
+                    ),
+                    # fmt: on
+                    # fmt: off
+                    # Wait for some of the files and the stop
+                    action.Repeated([action.WaitForOneOf([
+                        event.Start(0, FILES["testfile-bulk-01"].id),
+                        event.Start(0, FILES["testfile-bulk-02"].id),
+                        event.Start(0, FILES["testfile-bulk-03"].id),
+                        event.Start(0, FILES["testfile-bulk-04"].id),
+                        event.Start(0, FILES["testfile-bulk-05"].id),
+                        event.Start(0, FILES["testfile-bulk-06"].id),
+                        event.Start(0, FILES["testfile-bulk-07"].id),
+                        event.Start(0, FILES["testfile-bulk-08"].id),
+                        event.Start(0, FILES["testfile-bulk-09"].id),
+                        event.Start(0, FILES["testfile-bulk-10"].id),
+                    ])], 4),
+                    # fmt: on
+                    action.Stop(),
+                    # fmt: off
+                    # Wait of paused events
+                    action.WaitAndIgnoreExcept([
+                        event.Paused(0, FILES["testfile-bulk-01"].id),
+                        event.Paused(0, FILES["testfile-bulk-02"].id),
+                        event.Paused(0, FILES["testfile-bulk-03"].id),
+                        event.Paused(0, FILES["testfile-bulk-04"].id),
+                        event.Paused(0, FILES["testfile-bulk-05"].id),
+                        event.Paused(0, FILES["testfile-bulk-06"].id),
+                        event.Paused(0, FILES["testfile-bulk-07"].id),
+                        event.Paused(0, FILES["testfile-bulk-08"].id),
+                        event.Paused(0, FILES["testfile-bulk-09"].id),
+                        event.Paused(0, FILES["testfile-bulk-10"].id),
+                    ]),
+                    # fmt: on
+                    action.Sleep(2),
+                    action.ClearEventQueue(),
```

Comment:
Curious, what does this achieve?

Reply:
So, there might be some leftover events like `paused` or `started`. We don't know how many of them to expect since the `Stop()` can happen pretty independently of the receiver sending requests.